src/runtime/server/lib/oauth/microsoft.ts

```diff
@@ -133,24 +143,47 @@ export function microsoftEventHandler({ config, onSuccess, onError }: OAuthConfi
 
     const tokenType = tokens.token_type
     const accessToken = tokens.access_token
-    const userURL = config.userURL || 'https://graph.microsoft.com/v1.0/me'
+
     // TODO: improve typing
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    const user: any = await ofetch(userURL, {
-      headers: {
-        Authorization: `${tokenType} ${accessToken}`,
-      },
-    }).catch((error) => {
-      return { error }
-    })
-    if (user.error) {
-      const error = createError({
-        statusCode: 401,
-        message: `Microsoft login failed: ${user.error || 'Unknown error'}`,
-        data: user,
+    let user: any = {}
+
+    if (config.useUser) {
+      const userURL = config.userURL || 'https://graph.microsoft.com/v1.0/me'
+      user = await ofetch(userURL, {
+        headers: {
+          Authorization: `${tokenType} ${accessToken}`,
+        },
+      }).catch((error) => {
+        return { error }
       })
-      if (!onError) throw error
-      return onError(event, error)
+      if (user.error) {
+        const error = createError({
+          statusCode: 401,
+          message: `Microsoft login failed: ${user.error || 'Unknown error'}`,
+          data: user,
+        })
+        if (!onError) throw error
+        return onError(event, error)
+      }
+    }
+    else {
+      // use of any is required as MS has two token versions
+      // see: https://learn.microsoft.com/en-us/entra/identity-platform/access-tokens
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      const decodedToken = jwtDecode<any>(accessToken)
+      const msJwtVersion: '1.0' | '2.0' = decodedToken.ver
```

Comment:
That's a good point, I hadn't thought about that because currently there is no explicit validation of the access token, however the token is implicitly validated by calling "/me". 

I will implement the validation.

Reply:
Token validation is implemented if you want to take a look.